File: src/main/java/dark/leech/text/plugin/security/SecurityValidationException.java
package dark.leech.text.plugin.security;

/** Exception thrown when security validation fails. */
public class SecurityValidationException extends Exception {

    public SecurityValidationException(String message) {
        super(message);
    }

    public SecurityValidationException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: engine/src/commonMain/kotlin/dev/haipham22/leechtext/plugin/security/SecurityValidationException.kt
package dev.haipham22.leechtext.plugin.security

/** Exception khi security validation fail (port từ security/SecurityValidationException.java). */
class SecurityValidationException : Exception {
    constructor(message: String?) : super(message)
    constructor(message: String?, cause: Throwable?) : super(message, cause)
}

/** Exception khi cache capacity vượt limit (port từ security/CacheCapacityException.java). */
class CacheCapacityException(
    message: String?,
) : Exception(message)
